src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,167 @@
+
+    /**
+     * @return String of Horizontal Line
+     */
+    public String showLine() {
+        return HORIZONTAL_LINE;
+    }
+
+    /**
+     * Print error messages.
+     * @param errorMessage String error messages generated from Exceptions, etc...
+     */
+    public String showError(String errorMessage) {
+        return errorMessage;
+    }
+
+    /**
+     * Read the command input from console.
+     * @return String command in string form to be sent for parsing and executing.
+     */
+    public String readCommand() {
+        return scanner.nextLine();
+    }
+
+    public String readCommand(String command) {
+        return command;
+    }
+
+    /**
+     * Print ChatBot logo and greeting message.
+     */
+    public String showWelcome() {
+        String outputString = LOGO;
+        outputString += GREETINGS.get(1);
+        return outputString;
+    }
+
+    /**
+     * Close the initiated scanner and print farewell message.
+     */
+    public String showFarewell() {
+        this.scanner.close();
+        return FAREWELLS.get(1);
+    }
+
+    /**
+     * Print the numbered full list of task in TaskList.
+     * @param taskList TaskList to be displayed.
+     */
+    public String showTaskList(TaskList taskList) {
+        String outputString = this.showTaskListStatus(taskList);
+        int count = 1;
+        for (Task task : taskList) {
+            outputString += String.format("%d. %s\n", count, task);
+            count++;
+        }
+        return outputString;
+    }
+
+    /**
+     * Print success status when adding a task into TaskList.
+     * @param task Task to be added.
+     */
+    public String showAddedTask(Task task) {
+        return ("Got it. I've added this task: \n" + task.toString() + "\n");
+    }
+
+    /**
+     * Print the number of tasks inside the TaskList.
+     * @param taskList TaskList to be printed.
+     */
+    public String showTaskListStatus(TaskList taskList) {
+        return String.format("Now you have %d tasks in the list.\n", taskList.size());
+    }
+    /**
+     * Print success status when marking a task as Done status in TaskList.
+     * @param task Task to be marked as Done.
+     */
+    public String showMarkedTask(Task task) {
+        return ("Nice! I've marked this task as done: \n"
+                + task.toString() + "\n");
+    }
+    /**
+     * Print success status when unmarking a task into Undone status in TaskList.
+     * @param task Task to be marked as Undone.
+     */
+    public String showUnmarkedTask(Task task) {
+        return ("OK, I've marked this task as not done yet: \n"
+                + task.toString() + "\n");
+    }
+    /**
+     * Print success status when deleting a task from TaskList.
+     * @param task Task to be deleted.
+     */
+    public String showDeletedTask(Task task) {
+        return ("Noted. I've removed this task: \n"
+                + task.toString() + "\n");
+    }
```

Comment:
I liked your method naming convention, again they are very easy to differentiate and it explains them accurately!